pkg/cmd/pulumi/package_extract_schema.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/spf13/cobra"
+)
+
+func newExtractSchemaCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-schema <schema_source>",
```

Comment:
Wasn't there a proposal to offer CLI commands for invoking other plugin gRPC methods? If we're considering that, where would this command fit-in? cc: @Frassle

Reply:
That proposal, AFAIK, was to directly call the provider commands, and print the output. This command (while very similar) does a little more than that. It verifies the schema is valid, round tripping it through our parsed representation, then formatting the output.

Reply:
Agree with Ian, those commands are more intended for provider debugging, not for general user usage and while we would probably have a get-schema command under there as well for completeness it would just be a raw echo of the string sent back over grpc.